pyro/poutine/subsample_messenger.py

```diff
@@ -87,15 +87,15 @@ def _subsample(name, size=None, subsample_size=None, subsample=None, use_cuda=No
             assert subsample is None
             size = -1  # This is PyTorch convention for "arbitrary size"
             subsample_size = -1
-        elif subsample is None:
+        else:
             msg = {
                 "type": "sample",
                 "name": name,
                 "fn": _Subsample(size, subsample_size, use_cuda, device),
                 "is_observed": False,
                 "args": (),
                 "kwargs": {},
-                "value": None,
+                "value": subsample,
```

Comment:
Before this PR, if `subsample` was specified in the guide, no `_Subsample` site would be created, and hence the subsampling could not be traced and replayed. After this PR a `_Subsample` site is created whenever `size` is specified (and hence whenever subsampling is possible).